lib/saml_idp/name_id_formatter.rb

```diff
@@ -15,15 +18,16 @@ def all
     end
 
     def chosen
-      if split?
-        version, choose = "1.1", one_one.first
-        version, choose = "2.0", two_zero.first unless choose
-        version, choose = "2.0", "persistent" unless choose
-        build(version, choose)
-      else
-        choose = list.first || "persistent"
-        build("2.0", choose)
+      symbolized_name_id_format = sp_name_id_format.split(':').last.underscore.to_sym
+
+      unless list.key?(symbolized_name_id_format)
+        return build('2.0', list.select { |k, _| k == :persistent }.first)
```

Comment:
This is a change in default behavior. The original code would pick `list.first` by default, or `"persistent"` if the list was empty.

Reply:
The original code doesn't work and wasn't tested. If `config.name_id.formats` is not set in the IdP, it blows up with ` undefined method 'persistent' for #<User:0x00007ff0c251b930>`. Perhaps this should return 
```ruby
{
  name: "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent",
  getter: 'id'
}
```
as the default, since the Principal object is very likely to respond to `id`.